components/match2/commons/match_group_input.lua

```diff
@@ -904,6 +957,93 @@ function MatchGroupInput.setPlacement(opponents, winner, placementWinner, placem
 	return opponents
 end
 
+---@param match table
+---@param opponents {score: integer?}[]
+---@return boolean
+function MatchGroupInput.matchIsFinished(match, opponents)
+	local finished = Logic.readBoolOrNil(match.finished)
+	if finished ~= nil then
+		return finished
+	end
+
+	-- If a winner has been set
+	if Logic.isNotEmpty(match.winner) then
+		return true
+	end
+
+	-- If special status has been applied to a team
+	if MatchGroupInput.hasSpecialStatus(opponents) then
+		return true
+	end
+
+	if not MatchGroupInput.hasScore(opponents) then
+		return false
+	end
+
+	-- If enough time has passed since match started, it should be marked as finished
+	local threshold = match.dateexact and ASSUME_FINISHED_AFTER.EXACT or ASSUME_FINISHED_AFTER.ESTIMATE
+	if match.timestamp ~= DateExt.defaultTimestamp and (match.timestamp + threshold) < NOW then
+		return true
+	end
+
+	-- TODO: Investigate if bestof = 0 needs to be handled
```

Comment:
probably not needed but doing a simple ` or match.bestof == 0` probably wouldn't hurt

Reply:
Would be a breaking change if bestof == 0 return false, as it currently is true on wikis